### PR TITLE
feat: fogg create tfe folder

### PR DESCRIPTION
**Pull request by jakeyheath**
### Summary
This PR adds a feature to fogg that adds an optional tfe/ folder to synchronize fogg env/accounts with tfe workspaces.

### CODE REVIEW

File: apply/apply.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ import (
 const rootPath = "terraform"
 
 // Apply will run a plan and apply all the changes to the current repo.
-func Apply(fs afero.Fs, conf *v2.Config, tmp *templates.T, upgrade bool) error {
+func Apply(fs afero.Fs, conf *v2.Config, tmpl *templates.T, upgrade bool) error {
```
Comment (jakeyheath) [inline apply/apply.go:33]: tmp was confusing to me since it sounded like temporary

File: testdata/tfe_config/fogg.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  owner: foo@example.com
+  project: foo
+  providers: {}
+  terraform_version: 1.1.1
+envs:
+  comp1: {}
+  comp2: {}
+accounts:
+  account: {}
+tfe:
```
Comment (jakeyheath) [inline testdata/tfe_config/fogg.yml:15]: example of how to add this support to your fogg yml

File: testdata/tfe_config/fogg.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  terraform_version: 1.2.6
+  gh_org: chanzuckerberg
+  gh_repo: fogg
+  tfe_org: si.prod.tfe.czi.technology
+  module_source: github.com/chanzuckerberg/cztack//aws-params-reader-policy?ref=v0.15.1
```
Comment (jakeyheath) [inline testdata/tfe_config/fogg.yml:20]: in the real world, you would use something like "git@github.com:chanzuckerberg/shared-infra//terraform/modules/tfe-management?ref=tfe-management-v1.0.0"

Comment (jakeyheath) [reply testdata/tfe_config/fogg.yml:20]: I had to change it so CICD would pass since its a private module

**Review by alexlokshin-czi — APPROVED** [review]

**Pull request merged**